### PR TITLE
Update GraphQL schema documentation

### PR DESCRIPTION
**Pull request by pylipp**
I built the schema documentation that we host e.g. on api-staging.boxtribute.org/docs ([instructions](https://github.com/boxwise/boxtribute/blob/master/docs/graphql-api/README.md#usage)), and zipped it:
[docs.zip](https://github.com/user-attachments/files/16713998/docs.zip)

Extract and view `public/index.html`.

### CODE REVIEW

**Comment by pylipp** [comment]
[2.0 Improve documentation of graphQL schema](https://trello.com/c/9mky1CzF/1488-20-improve-documentation-of-graphql-schema)

File: back/boxtribute_server/business_logic/box_transfer/agreement/crud.py [hunk 1]
```diff
@@ -140,9 +140,9 @@ def create_transfer_agreement(
 ):
     """Insert information for a new TransferAgreement in the database. Update
     TransferAgreementDetail model with given source/target base information. By default,
-    the agreement is established between all bases of both organisations (indicated by
-    NULL for the Detail.source/target_base field). As a result, any base that added to
-    an organisation in the future would be part of such an agreement.
+    the agreement is established with all non-deleted bases of the partner organisation.
+    As a result, any base that added to the partner organisation will NOT be part of the
```
Comment (aerinsol) [inline back/boxtribute_server/business_logic/box_transfer/agreement/crud.py:144]: We should probably say and will need to establish an additional agreement to transfer goods

File: back/boxtribute_server/business_logic/user/fields.py [hunk 2]
```diff
@@ -8,7 +8,17 @@
 
 
 @user.field("bases")
-def resolve_bases(*_):
+def resolve_bases(user_obj, _):
+    if user_obj.id != g.user.id:
+        # If the queried user is different from the current user, we don't have a way
+        # yet to fetch secure information about that user's bases; it would require
```
Comment (aerinsol) [inline back/boxtribute_server/business_logic/user/fields.py:14]: So we will just throw an error or hit logins or...?

**Review by aerinsol — CHANGES_REQUESTED** [review]
Nice, just some small clarifications

File: back/boxtribute_server/business_logic/user/fields.py (hunk 2 above)
Comment (pylipp) [reply back/boxtribute_server/business_logic/user/fields.py:14]: we return `null` for the list of a user's bases

Comment (aerinsol) [reply back/boxtribute_server/business_logic/user/fields.py:14]: So basically it needs to be handled by the front end right? @HaGuesto or @fhenrich33 can you confirm it's been handled?

Comment (pylipp) [reply back/boxtribute_server/business_logic/user/fields.py:14]: yes but it's a field that the FE does not use (yet). Only user ID and name are used

**Pull request merged**